src/users/users.controller.ts

```diff
@@ -31,80 +35,71 @@ export class UsersController {
    * @throws 401 UNAUTHORIZED if the companyID requested is not one that the user has access to
    * @returns an array of Company objects that contain the companyID and associated User data
    */
-  @Get("users")
-  @Roles("breaktime-admin", "breaktime-supervisor")
-  public async getAllUsers(
+  @Get("usersCompanies")
+  public async getUsersAndCompanies(
     @Headers() headers: any,
+    @User() user: ValidatedUser,
     @Query("companyIds") companyIds?: string[],
-    @Query("roles") roles: string[] = ["associate"]
+    @Query("roles") roles: string[] = ["associate"],
+    @Query("userIds") userIds: string[] = []
   ): Promise<CompanyUsers[]> {
-    // TODO : This needs to get user role data as well, or we need to find a way to grab it from the Roles guard to check if a user is an admin or supervisor
-    const userId = await TokenClient.grabUserID(headers);
-    console.log(userId);
-    console.log(companyIds);
-
-    if (!userId) {
-      return [];
+    if (!user.sub) {
+      throw new HttpException(
+        "No authorized user found",
+        HttpStatus.UNAUTHORIZED
+      );
     }
 
-    // the company IDs the user belongs to as a supervisor
-    const userCompanyIds = (await GetCompaniesForUser(userId))
-      .SupervisorCompanyIDs;
+    //return this.userService.getUsers(user, companyIds ?? [], roles, userIds);
+    throw new Error("Not implemented.")
+  }  
 
-    // Get companyId(s) associated with user if no ids were provided via the queries
-    if (companyIds === undefined || companyIds.length === 0) {
-      companyIds = userCompanyIds;
-    } else {
-      // TODO: For later, we'll want to bypass this is a user is an admin
-      // Throw an error if the user is not an admin and doesn't have access to one or more of the companyIDs specified
-      for (const companyId of companyIds) {
-        if (!userCompanyIds.includes(companyId)) {
-          throw new HttpException(
-            `User is not authorized to access company data for company ID ${companyId}`,
-            HttpStatus.UNAUTHORIZED
-          );
-        }
-      }
+  @Get("userById")
```

Comment:
I think we can just rely on usersById instead of two endpoints doing the same thing.